ext/standard/proc_open.c

```diff
@@ -125,7 +125,7 @@ static php_process_env_t _php_array_to_envp(zval *environment, int is_persistent
 		if (key) {
 			l = ZSTR_LEN(key) + ZSTR_LEN(str) + 2;
 			memcpy(p, ZSTR_VAL(key), ZSTR_LEN(key));
-			strncat(p, "=", 1);
+			strncat(p, "=", 2);
```

Comment:
Why these changes? The man page says:

> If  src  contains n or more bytes, strncat() writes n+1 bytes to dest (n from src plus the terminating null byte).

Based on that I would expect that the string length needs to be passed, not the size with nul terminator.

Reply:
Because GCC 8 now raise a -Wstringop-overflow to notify possible bad usage (n+1 copied while only n "may" be expected)

Reply:
So in short, using 2 tells GCC: 'I know I'm going to use 2 bytes in dest.'

Reply:
If GCC does not like this code, then let's rewrite it with memcpy:

```
memcpy(p, ZSTR_VAL(key), ZSTR_LEN(key));
p[ZSTR_LEN(key)] = '=';
memcpy(p + ZSTR_LEN(key) + 1, ZSTR_VAL(str), ZSTR_LEN(str));
```

Using 2 for a string of length 1 is semantically wrong, imho.

Reply:
Partially fixed by 626bc0e7cc9c0ab3054de371ae5bbf33de4883e7 and 946a1e43149e95969caf721b2fbc1bbffc3c478c

This `strncpy` one is indeed very confusing, don't want to enter a bikesheding discussion, so closing this one, maybe some one will come back later on this.